documentation/CAMARA-OIDC-profile.md

```diff
@@ -100,7 +100,7 @@ The CAMARA profile uses the following OpenID Connect request parameters with the
 
 * **redirect_uri**
 
-  REQUIRED. Redirection URI to which the response will be sent. This URI MUST exactly match one of the Redirection URI values for the Client pre-registered at the OpenID Provider, with the matching performed as described in [RFC 3986](https://www.rfc-editor.org/rfc/rfc3986.html#section-6.2.1). When using this flow, the Redirection URI SHOULD use the https scheme; however, it MAY use the http scheme. The Redirection URI MAY use an alternate scheme, such as one that is intended to identify a callback into a native application.
+  REQUIRED. Redirection URI to which the response will be sent. This URI MUST exactly match one of the Redirection URI values for the Client pre-registered at the OpenID Provider, with the matching performed as described in [RFC 3986](https://www.rfc-editor.org/rfc/rfc3986.html#section-6.2.1). When using this flow, the Redirection URI MUST use the https scheme; however, it MAY use the http scheme only for non-production environments. The Redirection URI MAY use an alternate scheme, such as one that is intended to identify a callback into a native application.
```

Comment:
Please keep the text of the original standard, or delete this definition because it has not changed.
I think, this profile should not care what developers do in non-production. Also there are ways to have certificates on internal servers and developer machines e.g. self-signed certs and /etc/host entries.

Reply:
You are precisely saying that it should be referenced because there are no changes when in fact there are, or we want there to be. In another thread in this PR you said that the use of the `https` schema should be mandatory.

This is exactly what the [OIDC standard says](https://openid.net/specs/openid-connect-core-1_0.html#AuthRequest):

_redirect_uri
    REQUIRED. Redirection URI to which the response will be sent. This URI MUST exactly match one of the Redirection URI values for the Client pre-registered at the OpenID Provider, with the matching performed as described in Section 6.2.1 of [[RFC3986]](https://openid.net/specs/openid-connect-core-1_0.html#RFC3986) (Simple String Comparison). **When using this flow, the Redirection URI SHOULD use the https scheme; however, it MAY use the http scheme, provided that the Client Type is confidential, as defined in Section 2.1 of OAuth 2.0, and provided the OP allows the use of http Redirection URIs in this case. Also, if the Client is a native application, it MAY use the http scheme with localhost or the IP loopback literals 127.0.0.1 or [::1] as the hostname.** The Redirection URI MAY use an alternate scheme, such as one that is intended to identify a callback into a native application._

And what exactly the [OAuth2.0 standard says](https://datatracker.ietf.org/doc/html/rfc6749#section-3.1.2.1):

**This specification does not mandate the use of TLS** because at the time of this writing, requiring clients to deploy TLS is a significant hurdle for many client developers.

Reply:
What I meant with "don't change the standard" is that the text of a parameter definition should be verbatim the same as in the standard. 
We agreed in Camara that we use OIDC authorization code flow, so the OIDC standard is the basis of this PR.

Regarding "at this time":
> This specification does not mandate the use of TLS because at the time of this writing, requiring clients to deploy TLS is a significant hurdle for many client developers.

[OpenId Connect is now ten years old](https://self-issued.info/?p=2468) and OAuth2 even older. A lot has changed and improved.

The document is intended to be a profile of OIDC not a redefinition of OIDC based on OAuth2.
Please keep the OIDC definitions of parameters, if possible. If you change a definition, then please clearly mark the change.

I still think that just stating the differences to the profiled standards leads to a document that is easier to read and better for implementors. Please see https://github.com/AxelNennker/IdentityAndConsentManagement/blob/camara_oidc_profile/documentation/CAMARA-Security-Interoperability.md

Reply:
@AxelNennker brings up a lot of great points. I see a major issue with this and the demand for Data Minimization. @AxelNennker did you come across issues with the OIDC Profile and Data Minimization Regulation in your research?

Reply:
@garciasolero the definition of redirect_uri is different to the standard e.g. the part about confidential clients is not here instead there is the mention of non-production environments.

I read this definition as being more restrictive than the OIDC standard and I support that https is used for all client types and url-schemes for native apps. But maybe this is not the only interpretation. 
I think, changes to the standard should be clearly marked.